src/symbol/sprite_atlas.js

```diff
@@ -3,24 +3,20 @@
 const ShelfPack = require('@mapbox/shelf-pack');
 const browser = require('../util/browser');
 const util = require('../util/util');
+const window = require('../util/window');
+const Evented = require('../util/evented');
 
-class AtlasImage {
-    constructor(rect, width, height, sdf, pixelRatio) {
-        this.rect = rect;
-        this.width = width;
-        this.height = height;
-        this.sdf = sdf;
-        this.pixelRatio = pixelRatio;
-    }
-}
-
-class SpriteAtlas {
+// The SpriteAtlas class is responsible for turning a sprite and assorted
+// other images added at runtime into a texture that can be consumed by WebGL.
+class SpriteAtlas extends Evented {
 
     constructor(width, height) {
+        super();
+
         this.width = width;
         this.height = height;
 
-        this.atlas = new ShelfPack(width, height);
+        this.shelfPack = new ShelfPack(width, height);
```

Comment:
I see your point and agree that the semantics of the name are questionable. My intent was to reduce the size of the vocabulary and amount of indirection in this class.

 - `SpriteAtlas` has a `shelfPack`
 - `SpriteAtlas#shelfPack` is an instance of `ShelfPack`
 - `SpriteAtlas#shelfPack` has `bins`

-----

 - `SpriteAtlas` has a `atlas`
 - `SpriteAtlas#atlas` is an instance of `ShelfPack`
 - `SpriteAtlas#atlas` has `bins`